### PR TITLE
Feature/fluent datagrid

### PR DESCRIPTION
**Pull request by maxkatz6**
## What does the pull request do?

gif2

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

### CODE REVIEW

**Comment by maxkatz6** [comment]
@danwalmsley Fluent DataGrid uses named StaticResources that aren't updated after user changed theme. Probably it should be filled as a bug, that will be fixed with ThemeDictionaries or equivalent.

**Comment by maxkatz6** [comment]
TODO:
- [ ] Header GridLines doesn't work as expected
- [ ] DataGridRowHeader doesn't handle Row:pointerover
- [ ] For some reason Horizontal ScrollBar is visible, when it shouldn't
- [ ] Probably we should rename named static resourced used in DataGrid, make them more consistent

**Comment by maxkatz6** [comment]
@danwalmsley ready to review

File: src/Avalonia.Controls.DataGrid/Themes/Fluent.xaml [hunk 1]
```diff
@@ -0,0 +1,645 @@
...
+                <Grid x:Name="RootElement" />
+              </ControlTemplate>
+            </Grid.Resources>
+
+            <DataGridColumnHeader Name="PART_TopLeftCornerHeader"
+                                  Template="{StaticResource TopLeftHeaderTemplate}" />
+            <DataGridColumnHeadersPresenter Name="PART_ColumnHeadersPresenter"
+                                            Grid.Column="1"
+                                            Grid.ColumnSpan="2" />
+            <!--<DataGridColumnHeader Name="PART_TopRightCornerHeader"
```
Comment (danwalmsley) [inline src/Avalonia.Controls.DataGrid/Themes/Fluent.xaml:588]: remove commented code?

**Review by danwalmsley — CHANGES_REQUESTED** [review]
some left over commented code... however yet again another impressive PR ... good job! :)

**Review by danwalmsley — APPROVED** [review]
LGTM

**Comment by grokys** [comment]
@maxkatz6 I notice that this causes DevTools to display ugly black lines between columns:

Seems to be because the default theme defines:

```
<Setter Property="GridLinesVisibility" Value="Vertical" />
```

I'm guessing that grid lines were buggy before and so didn't work. The fluent theme sets this value to `None` - should we do that for the default theme too do you think?

**Comment by maxkatz6** [comment]
@grokys yes, grid lines used undefined brushes before. It was fixed with previous PR - https://github.com/AvaloniaUI/Avalonia/pull/4502/files#diff-c0b1938e2fc3f58e1ceda4a920cbd913L199-R201

Fluent theme has None because WCT has same value. And it looks good as for me.
Default theme with lines looks old-school for a bit though, especially with alternate row colors.

It's easy to change. I can create PR tomorrow if it will be better.

**Comment by grokys** [comment]
Already done ;) https://github.com/AvaloniaUI/Avalonia/pull/4572

**Pull request merged**